pandas/_typing.py

```diff
@@ -165,6 +170,50 @@ class IOargs(Generic[ModeVar, EncodingVar]):
 
     filepath_or_buffer: FileOrBuffer
     encoding: EncodingVar
-    compression: CompressionDict
-    should_close: bool
     mode: Union[ModeVar, str]
+    compression: CompressionDict
+    should_close: bool = False
+
+    def close(self) -> None:
+        """
+        Close the buffer if it was created by get_filepath_or_buffer.
+        """
+        if self.should_close:
+            assert not isinstance(self.filepath_or_buffer, str)
+            self.filepath_or_buffer.close()
+        self.should_close = False
+
+
+@dataclasses.dataclass
+class IOHandles:
+    """
+    Return value of io/common.py:get_handle
+
+    This is used to easily close created buffers and to handle corner cases when
+    TextIOWrapper is inserted.
+
+    handle: The file handle to be used.
+    created_handles: All file handles that are created by get_handle
+    is_wrapped: Whether a TextIOWrapper needs to be detached.
+    """
+
+    handle: Buffer
+    created_handles: List[Buffer] = dataclasses.field(default_factory=list)
+    is_wrapped: bool = False
+
+    def close(self) -> None:
```

Comment:
If there is actual *functionality* in here, shouldn't we rather move this to `io/common.py` (or somewhere in the io module) ? 
I would expect to have pure typing-related things in this file.

Reply:
okay, I will move it to `io/common`. Do you think it is preferable to then import `IOHandles` (and `IOArgs`) in `_typing` so that other modules can import `IOHandles` from `_typing` (there are only a few places that will need to import them) or should they directly import it from `io/common`?
@jorisvandenbossche @jbrockmendel

Reply:
yes ok to move it to io/common.py